src/middleware/user.middleware.ts

```diff
@@ -0,0 +1,78 @@
+import { Request, Response } from "express";
+import { StatusCodes } from "http-status-codes";
+
+import { UserService } from "../services";
+
+export class UserMiddleware {
+  constructor(private readonly userService: UserService) {}
+
+  autoSuggest() {
+    return (req: Request, res: Response): void => {
+      try {
+        const loginSubstring = req.query.loginSubstring as string;
+        const limit = parseInt(req.query.limit as string);
+        const users = this.userService.getAutoSuggestUsers(
+          loginSubstring,
+          limit
+        );
+        res.status(StatusCodes.OK).json(users);
+      } catch (err) {
+        res.status(StatusCodes.BAD_REQUEST).send(err);
+      }
+    };
+  }
+
+  getUserById() {
+    return (req: Request, res: Response): void => {
+      try {
+        const userId = req.params.id;
+        const user = this.userService.getUserByID(userId);
+        res.status(StatusCodes.OK).json(user);
+      } catch (err) {
+        res.status(StatusCodes.NOT_FOUND).send(err);
+      }
+    };
+  }
+
+  updateUser() {
+    return (req: Request, res: Response): void => {
+      try {
+        const userId = req.params.id;
+        const user = this.userService.updateUser(userId, req.body);
+        res.status(StatusCodes.OK).json(user);
+      } catch (err) {
+        res.status(StatusCodes.NOT_FOUND).send(err);
```

Comment:
I wouldn't send an error object here. Only status. res.sendStatus

Reply:
Changed. Thank you.